shared/feature/feeding/src/main/java/com/epmedu/animeal/feeding/presentation/viewmodel/handler/feeding/DefaultFeedingHandler.kt

```diff
@@ -55,29 +62,37 @@ class DefaultFeedingHandler(
     ErrorHandler by errorHandler {
 
     override var feedingStateFlow: StateFlow<FeedState> = stateFlow
+
+    private var fetchCurrentFeedingJob: Job? = null
+
     override fun CoroutineScope.fetchCurrentFeeding() {
-        launch {
+        fetchCurrentFeedingJob?.cancel()
+        fetchCurrentFeedingJob = launch {
             fetchCurrentFeedingPointUseCase()?.let { feedingPoint ->
-                updateState {
-                    copy(
-                        feedPoint = FeedingPointModel(feedingPoint),
-                        feedingConfirmationState = FeedingConfirmationState.Dismissed
-                    )
-                }
+                updateFeedingState(
+                    feedingConfirmationState = Dismissed,
+                    feedPoint = FeedingPointModel(feedingPoint)
+                )
                 showSingleReservedFeedingPoint(FeedingPointModel(feedingPoint))
                 startRoute()
             } ?: run {
-                updateState {
-                    copy(
-                        feedingConfirmationState = FeedingConfirmationState.Dismissed
-                    )
-                }
+                updateFeedingState(
+                    feedingConfirmationState = Dismissed
+                )
             }
         }
         stateFlow.onEach {
             // updates state flow at repository
             updateFeedStateUseCase(it.toDomainFeedState())
         }.launchIn(this)
+        launch {
+            getFeedStateUseCase().collect { feedState ->
+                updateFeedingState(
+                    feedingConfirmationState = feedState.feedingConfirmationState.toPresentationFeedingConfirmationState(),
+                    feedPoint = feedState.feedPoint?.let { FeedingPointModel(it) },
```

Comment:
If I remove this line, the state will be updated with `null` feeding point both in the handler and in the repository, which will lead to not started feeding on home screen.

Reply:
That is part of `updateFeedingState` , it's possible to use just
```
updateState{copy(
   feedingConfirmationState = ...
)}
```

Reply:
It's not, because we have to update feeding point from repository. Fetching feeding works only once after initialisation, but collect works until handler is destroyed or scope is canceled. "I will feed" and "Show on Map" rely on this `feeding point` field, not on `feedingConfirmationState`

Reply:
Yes, I could implement fetching feeding from server each time we open a screen, but Search and Favourites don't recreate as Home does, so I can't use `lifecycle.withCreated` for them